### DISCUSSION TITLE
Add node resolve and css plugins to rollup in extensions manager

### DISCUSSION THREAD

**Discussion by claytongulick** [directus/directus#14339] [Feature Requests]
There's a lot of work going on in improving DX, but one thing that stands out in day-to-day development is that creating and managing multiple extensions gets difficult when following the official extension development docs.

There's a case for "buildless" or "minimal" build app extensions - rollup is called on all extensions anyway, so it's very convenient to just have a simplified repo with extensions that are loaded at runtime by directus.

This doesn't work well currently because node_modules isn't resolved in the internal rollup() api call.

I've added https://www.npmjs.com/package/rollup-plugin-import-css and https://www.npmjs.com/package/@rollup/plugin-node-resolve to the extension manager, and it works like a charm with non SFC vue components, completely eliminating the need for an additional static build step.

Eventually, we'll probably want some way to have app extensions provide the build plugins they want, or a "rollup config" that gets used for the dynamic build, but for now just having node_modules an css available makes app extension development much easier.

PR incoming.

**Comment by claytongulick** [comment]
https://github.com/directus/directus/pull/14340

**Comment by benhaynes** [comment]
Quickly, what I'll mention is that our next big push is improving the developer experience and tooling around extension building. There have been other priorities to this point, but we're excited to be shifting into that side of the platform over the coming months.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
